arrow/arrow.py

```diff
@@ -405,12 +410,22 @@ def span_range(cls, frame, start, end, tz=None, limit=None, bounds="[)"):
         """
 
         tzinfo = cls._get_tzinfo(start.tzinfo if tz is None else tz)
-        start = cls.fromdatetime(start, tzinfo).span(frame)[0]
+        start = cls.fromdatetime(start, tzinfo).span(frame, exact=exact)[0]
+        end = cls.fromdatetime(end, tzinfo)
         _range = cls.range(frame, start, end, tz, limit)
-        return (r.span(frame, bounds=bounds) for r in _range)
+        for r in _range:
+            if exact:
+                floor, ceil = r.span(frame, bounds=bounds, exact=exact)
+                if ceil > end:
```

Comment:
The problem occurs here because we only check if `ceil > end`, however sometimes `floor==end` resulting in the bug in my example.

Reply:
@jadchaar I think if we get this bug fixed we're good to go here.